specreduce/wavelength_calibration.py

```diff
@@ -0,0 +1,248 @@
+        return self.refine(return_object=True)
+
+    def __str__(self):
+        return f"CalibrationLine: ({self.wavelength}, {self.pixel})"
+
+    def __repr__(self):
+        return f"CalibrationLine({self.wavelength}, {self.pixel})"
+
+
+class WavelengthCalibration1D():
+
+    def __init__(self, input_spectrum, lines, model=Linear1D(), spectral_unit=u.Angstrom,
+                 fitter=LMLSQFitter(calc_uncertainties=True),
+                 default_refinement_method=None, default_refinement_kwargs={}):
+        """
+        input_spectrum: `~specutils.Spectrum1D`
+            A one-dimensional Spectrum1D calibration spectrum from an arc lamp or similar.
+        lines: str, list
+            List of lines to anchor the wavelength solution fit. List items are coerced to
+            CalibrationLine objects if passed as tuples of (wavelength, pixel) with
+            default_refinement_method and default_refinement_kwargs as defaults.
+        model: `~astropy.modeling.Model`
+            The model to fit for the wavelength solution. Defaults to a linear model.
+        default_refinement_method: str, optional
+            words
+        default_refinement_kwargs: dict, optional
+            words
+        """
+        self.input_spectrum = input_spectrum
+        self.model = model
+        self.spectral_unit = spectral_unit
+        self.fitter = fitter
+        self.default_refinement_method = default_refinement_method
+        self.default_refinement_kwargs = default_refinement_kwargs
+
+        if self.default_refinement_method in ("gaussian", "min", "max"):
+            if 'range' not in self.default_refinement_kwargs:
+                raise ValueError("You must define 'range' in default_refinement_kwargs to use "
+                                 f"{self.refinement_method} refinement.")
+        elif (self.default_refinement_method == "gradient" and 'direction' not in
+                self.default_refinement_kwargs):
+            raise ValueError("You must define 'direction' in default_refinement_kwargs to use "
+                             "gradient refinement")
+
+        self.lines = []
+        if isinstance(lines, str):
+            if lines in self.available_line_lists:
+                raise ValueError(f"Line list '{lines}' is not an available line list.")
+        else:
+            for line in lines:
+                if isinstance(line, CalibrationLine):
+                    self.lines.append(line)
+                else:
+                    self.lines.append(CalibrationLine(self.input_spectrum, line[0], line[1],
+                                      self.default_refinement_method,
+                                      self.default_refinement_kwargs))
+
+    @classmethod
+    def autoidentify(cls, input_spectrum, line_list, model=Linear1D):
+        # line_list could be a string ("common stellar") or an object
+        # this builds a list of CalibrationLine objects and passes to __init__
+        return cls(...)
```

Comment:
remove from code for now?

Reply:
Would you prefer to do that, or raise a NotImplementedError?

Reply:
Fair, I'll just remove it.